examples/loki/nurf_data.py

```diff
@@ -102,120 +102,173 @@ def nurf_file_creator(loki_file, path_to_loki_file, data):
+                                           dtype=np.int32)
+                   
+        uv_inttime_data.attrs['long_name'] = 'uv_integration_time'
+        uv_inttime_data.attrs['units'] = 'us'  # TODO: unit to be verified, currently in micro-seconds
 
         # Fluorescence subgroup
         grp_fluo = hf.create_group("/entry/instrument/fluorescence")
-        grp_fluo.attrs["NX_class"] = 'NXdetector'
-        # grp_fluo.attrs["description"] = 'Fluorescence'
-        # grp_fluo.attrs["Fluo lightsource"] = 'LSM-265A,-310A with LDC-1'
-        grp_fluo.create_dataset('description', data='LSM-265A,-310A with LDC-1')
-
-        # create the various datasets for fluo
-        fluo_inttime = grp_fluo.create_dataset('fluo_integration_time',
-                                               data=data[
-                                                   'Fluo_IntegrationTime'],
-                                               shape=data[
-                                                   'Fluo_IntegrationTime'].shape,
+        grp_fluo.attrs["NX_class"] = 'NXdata'
+        
+        # currenty real fluo data is often messed up (i.e. empty spectra inbetween real ones)
+        # reshaping
+        data['Fluo_spectra']=np.reshape(data['Fluo_spectra'],(np.shape(data['Fluo_spectra'])[0],np.shape(data['Fluo_spectra'])[1]))
+        fluo_nb_spectra=np.shape(data['Fluo_spectra'])[0]
+        fluo_ik_spectra=np.zeros((1, fluo_nb_spectra))
+        
+        if data['Fluo_background'].ndim==1:
+            fluo_nb_dark=1
+        else:
+            fluo_nb_dark=np.shape(data['Fluo_background'])[1] #TODO: needs to be verified with Judith's setup
+        fluo_ik_dark=2*np.ones((1,fluo_nb_dark))
+        
+        if data['Fluo_intensity0'].ndim==1:
+            fluo_nb_ref=1
+        else:
+            fluo_nb_ref=np.shape(data['Fluo_intensity0'])[1] #TODO: needs to be verified with Judith's setup
+        fluo_ik_ref=4*np.ones((1,fluo_nb_ref))
+            
+        fluo_image_key=np.column_stack((fluo_ik_spectra,fluo_ik_dark, fluo_ik_ref))
+        
+        # Something is not okay with the real Fluo_intensity0 data from ILL. It contains only one 0, at least in my file from the ILL beamtime. 
+        # Also, some fluo spcetra in between are just dummy ones. There were acquisition problems.
+        # This code block can later be adjusted
+        try:
+            assert (np.shape(data['Fluo_intensity0'])[0]!=1), 'Fluo_intensity0 contains only one value.'
+        except AssertionError as error:
+            data['Fluo_intensity0']=data['Fluo_intensity0'][0]*np.ones((np.shape(data['Fluo_background'])[0]))
+           
+        # assemble all fluo data    
+        fluo_all_data=np.column_stack((data['Fluo_spectra'].T,data['Fluo_background'], data['Fluo_intensity0']))
+        
+        
+        # subgroup for fluo all data (sample, dark, reference)
+        fluo_signal=grp_fluo.create_group("fluo_all_data")
```

Comment:
Same problem as above: Why is there a subgroup with no Nexus class?

Reply:
Okay, I wanted to group the data with the keys.